compiler-java/src/com/redhat/ceylon/ceylondoc/CeylonDocTool.java

```diff
@@ -892,8 +892,7 @@ private void copySourceFiles() throws FileNotFoundException, IOException {
                 markup.around("script src='" + getResourceUrl(pkg, "ceylondoc.js") + "' type='text/javascript'"); 
                 markup.close("head");
                 markup.open("body", "pre data-language='ceylon' style='font-family: Inconsolata, Monaco, Courier, monospace'");
-                // XXX source char encoding
-                BufferedReader input = new BufferedReader(new InputStreamReader(pu.getUnitFile().getInputStream()));
+                BufferedReader input = new BufferedReader(new InputStreamReader(pu.getUnitFile().getInputStream(), getEncoding() != null ? getEncoding() : CeylonConfig.get(DefaultToolOptions.DEFAULTS_ENCODING)));
```

Comment:
Shouldn't there be a fallback, like this?

```ceylon
... : CeylonConfig.get(DefaultToolOptions.DEFAULTS_ENCODING, "utf8")));
```
On my system, `ceylon config get defaults.encoding` returns nothing.